test/unit/DADGenerator.spec.ts

```diff
@@ -0,0 +1,110 @@
+import 'reflect-metadata';
+import * as chai from 'chai';
+import * as sinon from 'sinon';
+
+import { IntegrationType } from '../../source/IIntegrationConfig';
+import DataAccessDoc from '../../source/DataAccess/DADGenerator';
+import * as fs from 'fs';
+
+const expect = chai.expect;
+
+describe('DADGenerator', () => {
+    describe('DataAccessDoc', () => {
+
+        describe('create()', () => {
+
+            let sandbox: sinon.SinonSandbox;
+            let dad;
+            let write;
+
+            beforeEach(() => {
+                dad = new DataAccessDoc(IntegrationType.Demo);
+                sandbox = sinon.createSandbox();
+                write = sandbox.stub(fs, 'writeFile');
+                write.returns(undefined);
+                sandbox.stub(dad, '_createTable');
+                dad._createTable.returns([
+                    'ROW 1',
+                    'ROW 2'
+                ]);
+            });
+
+            afterEach(() => {
+                sandbox.restore();
+            });
+
+            it('Should not return anything', () => {
+                expect(dad.create()).to.be.equal(undefined);
+            });
+
+            it('Should call createTable()', () => {
+                dad.create();
+                expect(dad._createTable.called).to.be.true;
+            });
+
+            it('Should call writeFile()', () => {
+                dad.create();
+                expect(write.called).to.be.true;
+            });
+
+        });
+
+        describe('_getQueryInfo()', () => {
+
+            let sandbox: sinon.SinonSandbox;
+            let dad;
+
+            beforeEach(() => {
+                dad = new DataAccessDoc(IntegrationType.Demo);
+                sandbox = sinon.createSandbox();
+                dad._queries = [{
+                    name: 'TABLE_NAME',
+                    query: 'SELECT * FROM TABLE_NAME'
+                }];
+            });
+
+            afterEach(() => {
+                sandbox.restore();
+            });
+
+            it('Should return an array of query objects: { tableName: string, fields: string[] }', () => {
+                const queryInfo = dad._getQueryInfo();
+                expect(queryInfo).to.be.an('array');
+                expect(queryInfo[0]).to.not.be.empty;
+                expect(queryInfo[0]).to.be.an('object');
+                expect(queryInfo[0].tableName).to.be.a('string');
+                expect(queryInfo[0].fields).to.be.an('array');
+            });
```

Comment:
Could expand with more tests to cover the regex operations taking place and validate that the queryInfos you get back contain the correct fields lists. You could take this up a notch and define a queryinfo interface with appropriate fields and tableName properties so that you can worry less about type checking and more about checking the values of the returned queryInfos

Reply:
Couldn't figure out a good implementation of an interface for just type checking (I'm sure there is one), but I think the .deep.equals(blah blah blah) should cover it.